plugins/maxruntime/maxruntime.go

```diff
@@ -0,0 +1,88 @@
+package maxruntime
+
+import (
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	schematypes "github.com/taskcluster/go-schematypes"
+	"github.com/taskcluster/taskcluster-worker/engines"
+	"github.com/taskcluster/taskcluster-worker/plugins"
+)
+
+type pluginProvider struct {
+	plugins.PluginProviderBase
+}
+
+type plugin struct {
+	plugins.PluginBase
+}
+
+type taskPlugin struct {
+	plugins.TaskPluginBase
+	maxRunTime int
+	done       chan bool
+	log        *logrus.Entry
+}
+
+type payloadType struct {
+	MaxRunTime int `json:"maxRunTime"`
+}
+
+var payloadSchema = schematypes.Object{
+	Properties: schematypes.Properties{
+		"maxRunTime": schematypes.Integer{
+			MetaData: schematypes.MetaData{
+				Title:       "Task execution timeout (ms)",
```

Comment:
I wonder if we should change this to seconds or minutes...
@petemoore What do think?

I think ms is very JS and JSON friendly... so I'm not oppose to it, but now is a great time to discuss this.